cookies/abtest_test.go

```diff
@@ -0,0 +1,76 @@
+package cookies
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestUnitABTest(t *testing.T) {
+
+	var testDomain = "www.ons.gov.uk"
+	var testTime = time.Date(2009, 11, 17, 20, 34, 58, 651387237, time.UTC)
+	var testServices = ABServices{NewSearch: &testTime}
+	var expectedEscapedCookieValue = "%7B%22new_search%22%3A%222009-11-17T20%3A34%3A58.651387237Z%22%7D"
+	var testMarshalledValue, _ = json.Marshal(testServices)
+	var testEscapedValue = url.QueryEscape(string(testMarshalledValue))
+
+	Convey("GetABTest", t, func() {
+		Convey("returns cookie value if value is set", func() {
+			req := httptest.NewRequest("GET", "/", nil)
+			req.AddCookie(&http.Cookie{Name: aBTestKey, Value: testEscapedValue})
+			cookie, _ := GetABTest(req)
```

Comment:
It is always worth checking error is nil value
```suggestion
			cookie, err := GetABTest(req)
			So(err, ShouldBeNil)
```

Reply:
👍